<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MetricsWidget</class>
 <widget class="QWidget" name="MetricsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>300</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Performance Metrics</string>
  </property>
  <layout class="QGridLayout" name="mainLayout">
   <property name="spacing">
    <number>5</number>
   </property>
   <property name="leftMargin">
    <number>5</number>
   </property>
   <property name="topMargin">
    <number>5</number>
   </property>
   <property name="rightMargin">
    <number>5</number>
   </property>
   <property name="bottomMargin">
    <number>5</number>
   </property>
   <item row="0" column="0">
    <widget class="QGroupBox" name="performanceGroup">
     <property name="title">
      <string>Performance</string>
     </property>
     <layout class="QGridLayout" name="performanceLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="fpsLabel">
        <property name="text">
         <string>FPS:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="fpsValue">
        <property name="text">
         <string>0.0</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="updateTimeLabel">
        <property name="text">
         <string>Update Time:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="updateTimeValue">
        <property name="text">
         <string>0.00 ms</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="collisionTimeLabel">
        <property name="text">
         <string>Collision Time:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="collisionTimeValue">
        <property name="text">
         <string>0.00 ms</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="memoryLabel">
        <property name="text">
         <string>Memory:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="memoryValue">
        <property name="text">
         <string>0.0 MB</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QProgressBar" name="memoryBar">
        <property name="maximum">
         <number>1000</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="trafficGroup">
     <property name="title">
      <string>Traffic Flow</string>
     </property>
     <layout class="QGridLayout" name="trafficLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="totalVehiclesLabel">
        <property name="text">
         <string>Total Vehicles:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="totalVehiclesValue">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="activeVehiclesLabel">
        <property name="text">
         <string>Active:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="activeVehiclesValue">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="queuedVehiclesLabel">
        <property name="text">
         <string>Queued:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="queuedVehiclesValue">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="avgSpeedLabel">
        <property name="text">
         <string>Avg Speed:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="avgSpeedValue">
        <property name="text">
         <string>0.0</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="throughputLabel">
        <property name="text">
         <string>Throughput:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLabel" name="throughputValue">
        <property name="text">
         <string>0.0 /min</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="intersectionGroup">
     <property name="title">
      <string>Intersection</string>
     </property>
     <layout class="QGridLayout" name="intersectionLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="maxQueueLabel">
        <property name="text">
         <string>Max Queue:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="maxQueueValue">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="totalQueueLabel">
        <property name="text">
         <string>Total Queue:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="totalQueueValue">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="congestionLabel">
        <property name="text">
         <string>Congestion:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="congestionValue">
        <property name="text">
         <string>0%</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QProgressBar" name="congestionBar">
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="avgWaitLabel">
        <property name="text">
         <string>Avg Wait:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLabel" name="avgWaitValue">
        <property name="text">
         <string>0.0 s</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QGroupBox" name="systemGroup">
     <property name="title">
      <string>System</string>
     </property>
     <layout class="QGridLayout" name="systemLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="gridEfficiencyLabel">
        <property name="text">
         <string>Grid Efficiency:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="gridEfficiencyValue">
        <property name="text">
         <string>0%</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="occupiedCellsLabel">
        <property name="text">
         <string>Occupied Cells:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="occupiedCellsValue">
        <property name="text">
         <string>0/132</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QProgressBar" name="gridEfficiencyBar">
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>